src/test/java/it/pagopa/pn/cucumber/steps/recipient/RicezioneNotificheWebSteps.java

```diff
@@ -751,4 +764,230 @@ private static void waitState(Integer waitingStateCsv) {
+                boolean exists = legalAddressByRecipient.stream()
+                        .anyMatch(address -> LegalChannelType.PEC.equals(address.getChannelType()) && senderId.equals(address.getSenderId()));
+
+                Assertions.assertTrue(exists, "PEC NOT FOUND");
+            }
+        });
+    }
+
+    @And("viene rimossa se presente la pec per il comune {string}")
+    public void vieneRimossaSePresenteLaPecDiPiattaformaDi(String pa) {
+        String senderId = getSenderIdPa(pa);
+        Assertions.assertDoesNotThrow(() -> {
+            this.iPnWebUserAttributesClient.deleteRecipientLegalAddress(senderId, LegalChannelType.PEC);
+            log.info("PEC FOUND AND DELETED");
+        }, "PEC NOT FOUND");
+    }
+
+    @And("vengono rimossi eventuali recapiti presenti per l'utente")
+    public void cleanLegalAddressForUser() {
+        try {
+            List<LegalAndUnverifiedDigitalAddress> legalAddressByRecipient = this.iPnWebUserAttributesClient.getAddressesByRecipient().getLegal();
+            if (legalAddressByRecipient != null && !legalAddressByRecipient.isEmpty()) {
+                legalAddressByRecipient.stream()
+                        .forEach(address -> {
+                            this.iPnWebUserAttributesClient.deleteRecipientLegalAddress(address.getSenderId(), address.getChannelType());
+                            log.info("Cancellato indirizzo di tipo " + address.getChannelType() + " per il comune " + address.getSenderId());
+                        });
+            }
+            List<CourtesyDigitalAddress> courtesyDigitalAddresses = this.iPnWebUserAttributesClient.getAddressesByRecipient().getCourtesy();
+            if (courtesyDigitalAddresses != null && !courtesyDigitalAddresses.isEmpty()) {
+                courtesyDigitalAddresses.stream()
+                        .forEach(address -> {
+                            this.iPnWebUserAttributesClient.deleteRecipientCourtesyAddress(address.getSenderId(), address.getChannelType());
+                            log.info("Cancellato indirizzo di cortesia di tipo " + address.getChannelType() + " per il comune " + address.getSenderId());
+                        });
+            }
+        } catch (Exception e) {
+            log.error("RIMOZIONE RECAPITI FALLITA:\n" + e);
+        }
+    }
+
+    @And("Viene verificato che non sia arrivato un evento di {string}")
+    public void verificaAssenzaElementoTimeline(String categoryToFind) {
+        FullSentNotificationV23 fullSentNotificationV23 = sharedSteps.getSentNotification();
+        List<it.pagopa.pn.client.b2b.pa.generated.openapi.clients.externalb2bpa.model.TimelineElementV23> timelineElements = fullSentNotificationV23.getTimeline();
+        boolean exist = timelineElements.stream()
+                .anyMatch(element -> element.getCategory() != null && element.getCategory().toString().equals(categoryToFind));
+        Assertions.assertFalse(exist);
+    }
```

Comment:
Il metodo sotto può essere semplificato come suggerito
```suggestion
    @And("Viene verificato che non sia arrivato un evento di {string}")
    public void verificaAssenzaElementoTimeline(String categoryToFind) {
        sharedSteps.getSentNotification().getTimeline()
            .stream()
            .map(it.pagopa.pn.client.b2b.pa.generated.openapi.clients.externalb2bpa.model.TimelineElementV23::getCategory)
            .filter(Objects::nonNull)
            .map(it.pagopa.pn.client.b2b.pa.generated.openapi.clients.externalb2bpa.model.TimelineElementCategoryV23::toString)
            .filter(category -> category.equals(categoryToFind))
            .findAny()
            .ifPresentOrElse(null, () -> { throw new AssertionFailedError("L'evento cercato è stato ritornato!");});
    }
```

Reply:
il metodo è stato modificato come suggerito